File: backend/doc-core-service/src/main/java/com/docgen/core/domain/model/Document.java
package com.docgen.core.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Document {
    private String id;
    private String projectId;
    private String title;
    private String content; // Markdown or structured content
    private DocumentType type;
    private DocumentStatus status;
    private List<Requirement> requirements;
    private LocalDateTime generatedAt;
    
    public enum DocumentType {
        TECHNICAL_SPEC,
        API_DOCS,
        README,
        ARCHITECTURE_DIAGRAM
    }
    
    public enum DocumentStatus {
        PENDING,
        GENERATING,
        COMPLETED,
        FAILED
    }
}

File: frontend/client/src/domain/models/Document.ts
import { Requirement } from "./Requirement";

export enum DocumentType {
  TECHNICAL_SPEC = "TECHNICAL_SPEC",
  API_DOCS = "API_DOCS",
  README = "README",
  ARCHITECTURE_DIAGRAM = "ARCHITECTURE_DIAGRAM",
}

export enum DocumentStatus {
  PENDING = "PENDING",
  GENERATING = "GENERATING",
  COMPLETED = "COMPLETED",
  FAILED = "FAILED",
}

export interface Document {
  id: string;
  projectId: string;
  title: string;
  content: string;
  type: DocumentType;
  status: DocumentStatus;
  requirements: Requirement[];
  generatedAt?: string;
}
